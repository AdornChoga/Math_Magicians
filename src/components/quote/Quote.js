import './Quote.css';

const Quote = () => (
  <div className="quote-container">
    <p>
      Mathematics is not about numbers, equations, computation, or
      algorithms. It is about understanding. -William Paul Thurston
    </p>
  </div>
);

export default Quote;
