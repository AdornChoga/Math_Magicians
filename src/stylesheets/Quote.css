* {
  box-sizing: border-box;
}

html {
  font-size: 62.5%;
}

body {
  padding: 0;
  margin: 0;
}

.quote-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  padding: 0 20rem;
}

.quote-container p {
  font-size: 2rem;
  font-family: sans-serif;
  font-weight: bold;
  outline: 0.4rem inset #f5913e;
  padding: 5rem;
}
