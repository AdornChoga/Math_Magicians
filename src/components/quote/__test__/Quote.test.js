import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quote from '../Quote';

afterEach(cleanup);

describe('Tests for Quote rendering and snapshot matching.', () => {
  test('Is Quote rendered?', () => {
    render(<Quote />);
    const quoteElement = screen.getByTestId('quote-page');
    expect(quoteElement).toBeInTheDocument();
  });

  test('Is Quote snapshot created/matched?', () => {
    const tree = renderer.create(<Quote />);
    expect(tree).toMatchSnapshot();
  });
});
