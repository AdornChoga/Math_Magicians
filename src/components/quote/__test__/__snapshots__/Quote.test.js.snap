// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Quote rendering and snapshot matching. Is Quote snapshot created/matched? 1`] = `
<div
  className="quote-container"
  data-testid="quote-page"
>
  <p>
    Mathematics is not about numbers, equations, computation, or algorithms. It is about understanding. -William Paul Thurston
  </p>
</div>
`;
